### ISSUE TITLE
Write paper

### ISSUE THREAD

**Issue by cjacoby** [ejhumphrey/harmonic-cnn#28]
* Write the paper
* Get publish it
* ... profit

**Comment by ejhumphrey** [comment]
more seriously, do we think we can agree on a final(ish) outline on this guy so we can divide and conquer, rather than continuing to ramble blindly?

a proposal

1. intro 

  i. deep learning, so hot right now
  ii. convnets, such awesome
  iii. but they don't make sense for audio
  iv. we propose HCNNs, which exploit this structure, present a hybrid dataset for instrument classification, and train / evaluate accordingly

2. What are HCNNs? + prior art

  i. Lostanlen, ISMIR2016 ... others? maybe WaveNet now?
  ii. CQT Spectra, how do we fold it
  iii. rest of the architecture

3. Method
  i. MINST as data source - what and why?
  ii. Model parameters and configurations
  iii. Training strategy (partitions, presentation) and evaluation

4. Results / Discussion

**Comment by ejhumphrey** [comment]
bump @cjacoby @bmcfee

**Comment by cjacoby** [comment]
Section about Harmonic correlation in music (maybe this is already 2.ii?), like that plot you were making a while back?

In general, LGTM, tho.

**Comment by bmcfee** [comment]
Good point on wavenet -- dilated convolutions in the time domain are explicitly modeling harmonics.  I hadn't thought of that.  Not sure it's worth citing at this point though, since last i checked, their paper was still locked down by a google wall.

I think there's a nice figure to be drawn that would look like:
```
[x]\
[x]\\
[x]--- w  (dense / full-height)
[x]//
[x]/

[ ]
[x]\
[x]-- w  (sparse / frequency-local)
[x]/
[ ]

[x]\
[ ] \
[x]-- w  (sparse / harmonic-local)
[ ] /
[x]/
```

but, you know, with colors and boxes and more than five bins.

**Comment by ejhumphrey** [comment]
@cjacoby, that was going to be in 1.iii, but maybe it's better in 2.i/ii

@bmcfee, maybe worth forward citing? I can't think of any other relevant
work. Also, can has help articulating differences between this and
Lostalen's ISMIR2016 work? Diagram is a great idea btw.

**Comment by bmcfee** [comment]
> maybe worth forward citing? I can't think of any other relevant
> work.

i'm not so sure about forward-citing a paper that none of us have/can read yet, since the odds of mischaracterizing their method seem pretty high.



> Also, can has help articulating differences between this and
> Lostalen's ISMIR2016 work? Diagram is a great idea btw.

I can help with this.  I'll get to work on the figure now that experiments are running on mariana.

**Comment by ejhumphrey** [comment]
I read this paper?
https://drive.google.com/file/d/0B3cxcnOkPx9AeWpLVXhkTDJINDQ/view

It's not arxiv, but it's not far away from that (and might be there
already?).

**Comment by ejhumphrey** [comment]
Also worth mentioning, I won't be able to put cycles against this until
maybe 8am CET tomorrow (2am EST/11pm PST), so please commit / push before
punching out and I'll pick up where ya leave off.

**Comment by cjacoby** [comment]
I'm going to try to come up with a clean way of publishing the analysis for
each model into the github repo so you can work with it there. I think that
code is 75% done, but I haven't looked recently.

**Comment by ejhumphrey** [comment]
👍

**Comment by bmcfee** [comment]
@ejhumphrey you want me to push on #41 for that, or make a separate pr?

**Comment by ejhumphrey** [comment]
how bout we sail 41 through (unless anyone thinks it's complete rubbish) and then work from there? 

I just brought it up to speed with master so nothing should fail this time around (it's just latex after all).

**Comment by bmcfee** [comment]
i haven't read it yet, but sure

**Comment by ejhumphrey** [comment]
it's .... not my best work.

**Comment by bmcfee** [comment]
![image](https://cloud.githubusercontent.com/assets/1190540/18398894/b549324c-769c-11e6-8adb-0137ce0136a8.png)

how's this look?

**Comment by ejhumphrey** [comment]
Thumbs up from this guy

**Comment by bmcfee** [comment]
you want to merge #41 and then i'll start hacking this stuff in?

**Comment by ejhumphrey** [comment]
Can't now, feel free to pull the trigger

**Comment by ejhumphrey** [comment]
Also what's up with that Travis build? 23+ minutes??

**Comment by cjacoby** [comment]
Just ignore it for now; things pass locally in half the time. I'll try to
shorten it tonight or tomorrow after all the more important stuff is done.
The integration test actually covers all the important stuff but is not
getting run by Travis right now.
